runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java

```diff
@@ -165,4 +217,38 @@ else if (client.getGameState() == GameState.LOGGED_IN && (force || !loggedIn))
 			discordState.triggerEvent(DiscordGameEventType.IN_GAME, config.actionDelay());
 		}
 	}
+
+	private void triggerUpdate()
+	{
+		if (lastSkillEvent != null && config.showSkillingActivity())
+		{
+			lastEvent.setState(lastSkillEvent.getState());
```

Comment:
Dont mutate basically constants (enum == constants). Store the lastEvent.getState and lastEvent.getImageKey separately.